### PR TITLE
fix: kubernetes installation, high availability and publish to internet docs based on usability testing findings

### PR DESCRIPTION
**Pull request by jnikhila**
## Checklist
I have:
- [x] run the content through Grammarly
- [ ] linked to sample apps when relevant
- [x] added the meta description for each page in the PR
- [ ] minimized the callouts and added only when necessary
- [ ] added the `queryString` parameter to the Tabs (if used)
- [ ] masked PII in images. For example, login credentials, account details, and more
- [ ] added images only when necessary
- [ ] deleted the images that are no longer used for the updated pages in the PR
- [ ] followed the image file naming convention while renaming or adding new images. (Use lowercase letters, dashes between words, and be as descriptive as possible)
- [ ] used the `<figure/>` tag instead of a markdown representation for images
- [ ] added the `<figcaption/>` tag to add a caption to the image
- [ ] added the `alt` attribute in the `` tag
- [ ] verified and updated the cross-references or created redirect rules for the changed or removed content 
- [x] reviewed and applied the style changes for UI formatting. For example, Bold the UI elements(Buttons on screen) used in the doc.

### CODE REVIEW

File: website/docs/getting-started/setup/installation-guides/kubernetes/configure-high-availability.mdx [hunk 1]
```diff
@@ -12,128 +12,147 @@ High Availability is only available on Kubernetes self-hosted [**Business Editio
...
+      enabled: false
+      service:
+        nameOverride: appsmith-mongodb
+    ```
+  
+* Update the below parameters to point to an external MongoDB:
+
+    ```yaml
+    # configure an external MongoDB
+      applicationConfig::
```
Comment (pratapaprasanna) [inline website/docs/getting-started/setup/installation-guides/kubernetes/configure-high-availability.mdx:114]:
```suggestion
      applicationConfig:
```

File: website/docs/getting-started/setup/installation-guides/kubernetes/configure-high-availability.mdx [hunk 2]
```diff
@@ -12,128 +12,147 @@ High Availability is only available on Kubernetes self-hosted [**Business Editio
...
+        enabled: false
+        auth:
+          enabled: false
+    ```
+  
+* Update the below parameters to point to an external Redis:
+
+    ```yaml
+    # configure an external Redis
+      applicationConfig::
```
Comment (pratapaprasanna) [inline website/docs/getting-started/setup/installation-guides/kubernetes/configure-high-availability.mdx:134]:
```suggestion
      applicationConfig:
```

File: website/docs/getting-started/setup/installation-guides/kubernetes/README.mdx [hunk 3]
```diff
@@ -11,35 +11,50 @@ This page provides steps to install Appsmith on a Kubernetes cluster using the [
...
 
-  a. Set up Kubernetes by [working with kubectl](https://microk8s.io/docs/working-with-kubectl).
+    ```bash
+    kubectl apply -f - <<EOF
+      apiVersion: storage.k8s.io/v1
+      kind: StorageClass
+      apiVersion: storage.k8s.io/v1
+      metadata:
+        name: gp2
+      provisioner: kubernetes.io/aws-ebs
```
Comment (pratapaprasanna) [inline website/docs/getting-started/setup/installation-guides/kubernetes/README.mdx:53]:
```suggestion
      provisioner: kubernetes.io/aws-ebs
      EOF
```

File: website/docs/getting-started/setup/installation-guides/kubernetes/configure-high-availability.mdx [hunk 4]
```diff
@@ -12,128 +12,147 @@ High Availability is only available on Kubernetes self-hosted [**Business Editio
...
+    minReplicas: 2
+    maxReplicas: 2
   ```
 
-  b. Using existing PVC: this is useful if you already have a PVC that you want to reuse for EFS or if you have multiple deployments or pods that need access to the same EFS volume. Update the `values.yaml` file under the `persistence` attribute as shown below:
+2. You may choose to configure external PostgreSQL, MongoDB, and Redis instances for your self-hosted instance Follow these steps to configure an external PostgreSQL, MongoDB, and Redis instances:
+  * Turn off the settings for internal PostgreSQL, by changing the `enabled` attribute to `false` in the `values.yaml` file for postgresql attribute.
+  
+    ```yaml
+    # enable PostgreSQL
```
Comment (pratapaprasanna) [inline website/docs/getting-started/setup/installation-guides/kubernetes/configure-high-availability.mdx:80]: lets remove this line
```
# enable PostgreSQL
```

File: website/docs/getting-started/setup/installation-guides/kubernetes/configure-high-availability.mdx [hunk 5]
```diff
@@ -12,128 +12,147 @@ High Availability is only available on Kubernetes self-hosted [**Business Editio
...
+      APPSMITH_KEYCLOAK_DB_DRIVER: "postgresql"
+      APPSMITH_KEYCLOAK_DB_USERNAME: "POSTGRES_USER"
+      APPSMITH_KEYCLOAK_DB_PASSWORD: "POSTGRES_PASSWORD"
+      APPSMITH_KEYCLOAK_DB_URL: "POSTGRES_DB_URL"
+    ```
+
+* Turn off the settings for internal MongoDB, by changing the `enabled` attribute to `false` in the `values.yaml` file for mongodb attribute.
+
+    ```yaml
+    # enable MongoDB
```
Comment (pratapaprasanna) [inline website/docs/getting-started/setup/installation-guides/kubernetes/configure-high-availability.mdx:102]: lets remove 
```
enable mongodb
```

File: website/docs/getting-started/setup/installation-guides/kubernetes/configure-high-availability.mdx [hunk 6]
```diff
@@ -12,128 +12,147 @@ High Availability is only available on Kubernetes self-hosted [**Business Editio
...
+    # configure an external MongoDB
+      applicationConfig:
+        #highlight-next-line
+        APPSMITH_MONGODB_URI: "MONGODB_URI"
+    ```
+
+* Turn off the settings for internal Redis, by changing the `enabled` attribute to `false` in the `values.yaml` file for redis attribute.
+
+    ```yaml
+      # enable Redis
```
Comment (pratapaprasanna) [inline website/docs/getting-started/setup/installation-guides/kubernetes/configure-high-availability.mdx:122]: Lets remove this line
```
enable redis
```

File: website/docs/getting-started/setup/installation-guides/kubernetes/README.mdx [hunk 7]
```diff
@@ -11,35 +11,51 @@ This page provides steps to install Appsmith on a Kubernetes cluster using the [
...
 
-<Tabs groupId="configurations" queryString="current-configuration">
- <TabItem label="AWS EKS" value="AWSEKS">
+1. At least 15 GB of free storage space.
+2. 4 GB of RAM.
+3. Installed Helm package manager on your local machine. See the official Helm documentation for your [operation system](https://helm.sh/docs/intro/install/#through-package-managers).
+4. A running Kubernetes cluster. If not present, follow these steps to set up the cluster:
 
-  a. Set up the Kubernetes cluster by [creating a kubeconfig for Amazon EKS](https://docs.aws.amazon.com/eks/latest/userguide/create-kubeconfig.html).
+    a. Verify if you have access to AWS EKS with:
```
Comment (pratapaprasanna) [inline website/docs/getting-started/setup/installation-guides/kubernetes/README.mdx:22]:
```suggestion
    a. Verify if you have access to AWS cli with:
```

**Review by pratapaprasanna — APPROVED** [review]

File: website/docs/getting-started/setup/installation-guides/kubernetes/README.mdx [hunk 8]
```diff
@@ -11,72 +11,97 @@ This page provides steps to install Appsmith on a Kubernetes cluster using the [
...
+    ```
+    The above command provides a summary of the current cluster configuration, including the Kubernetes master and other cluster information. 
 
-</TabItem>
-<TabItem label="Microk8s" value="Microk8s">
+    d. A default storage class running. If not created yet, create the storage class with:
 
-  a. Set up Kubernetes by [working with kubectl](https://microk8s.io/docs/working-with-kubectl).
+    ```bash
+    kubectl apply -f - <<EOF
```
Comment (jnikhila) [inline website/docs/getting-started/setup/installation-guides/kubernetes/README.mdx:47]: @pratapaprasanna - Please update the storage file command based on our discussion.

File: website/docs/getting-started/setup/installation-guides/kubernetes/README.mdx [hunk 9]
```diff
@@ -11,72 +11,99 @@ This page provides steps to install Appsmith on a Kubernetes cluster using the [
...
-4. Follow the steps below to install Appsmith on your Kubernetes cluster:
-
-<Tabs groupId="configurations" queryString="current-configuration">
- <TabItem label="AWS EKS" value="AWSEKS">
-
-  a. Set up the Kubernetes cluster by [creating a kubeconfig for Amazon EKS](https://docs.aws.amazon.com/eks/latest/userguide/create-kubeconfig.html).
-
-  b. Ensure that you have a default storage class running. If not, [create default storage class](https://docs.aws.amazon.com/eks/latest/userguide/storage-classes.html).
+## Prerequisites
+1. At least 15 GB of free storage space.
```
Comment (danciaclara) [inline website/docs/getting-started/setup/installation-guides/kubernetes/README.mdx:15]: these can be just bullets

File: website/docs/getting-started/setup/installation-guides/kubernetes/README.mdx [hunk 10]
```diff
@@ -11,72 +11,99 @@ This page provides steps to install Appsmith on a Kubernetes cluster using the [
...
+2. 4 GB of RAM.
+
+## Before you begin
+1. Installed Helm package manager on your local machine. See the official Helm documentation for your [operating system](https://helm.sh/docs/intro/install/#through-package-managers).
```
Comment (danciaclara) [inline website/docs/getting-started/setup/installation-guides/kubernetes/README.mdx:19]:
```suggestion
1. Install Helm package manager on your local machine. See the official Helm documentation for your [operating system](https://helm.sh/docs/intro/install/#through-package-managers).
```

File: website/docs/getting-started/setup/installation-guides/kubernetes/README.mdx [hunk 11]
```diff
@@ -11,72 +11,99 @@ This page provides steps to install Appsmith on a Kubernetes cluster using the [
...
+2. Installed and configured `kubectl` to interact with your Kubernetes cluster. Follow the below guides available on the official Kubernetes documentation for instructions on how to install `kubectl` on your specific operating system:
```
Comment (danciaclara) [inline website/docs/getting-started/setup/installation-guides/kubernetes/README.mdx:20]:
```suggestion
2. Install and configure `kubectl` to interact with your Kubernetes cluster. Follow the below guides available on the official Kubernetes documentation for instructions on how to install `kubectl` on your specific operating system: 
```

File: website/docs/getting-started/setup/installation-guides/kubernetes/README.mdx [hunk 12]
```diff
@@ -11,72 +11,99 @@ This page provides steps to install Appsmith on a Kubernetes cluster using the [
...
+1. At least 15 GB of free storage space.
+2. 4 GB of RAM.
+
+## Before you begin
+1. Installed Helm package manager on your local machine. See the official Helm documentation for your [operating system](https://helm.sh/docs/intro/install/#through-package-managers).
+2. Installed and configured `kubectl` to interact with your Kubernetes cluster. Follow the below guides available on the official Kubernetes documentation for instructions on how to install `kubectl` on your specific operating system: 
+    * [MacOS using Homebrew](https://kubernetes.io/docs/tasks/tools/install-kubectl-macos/#install-with-homebrew-on-macos)
+    * [Linux using native package management](https://kubernetes.io/docs/tasks/tools/install-kubectl-linux/#install-using-native-package-management)
+    * [Windows using Chocolatey, Scoop, or winget](https://kubernetes.io/docs/tasks/tools/install-kubectl-windows/#install-nonstandard-package-tools).
+3. A running Kubernetes cluster. If not present, follow these steps to set up the cluster on AWS EKS:
```
Comment (danciaclara) [inline website/docs/getting-started/setup/installation-guides/kubernetes/README.mdx:24]:
```suggestion
3. Set up a Kubernetes cluster. Follow these steps to set up the cluster on AWS EKS:
```

File: website/docs/getting-started/setup/installation-guides/kubernetes/README.mdx [hunk 13]
```diff
@@ -11,72 +11,99 @@ This page provides steps to install Appsmith on a Kubernetes cluster using the [
...
+
+  :::info
+   You are free to choose your preferred platform for hosting the Kubernetes cluster. Make sure you have configured Kubeconfig and created a default storage class before proceeding with the Appsmith installation. If you face any issues, reach out to the support team at [support@appsmith.com](mailto:support@appsmith.com).
```
Comment (danciaclara) [inline website/docs/getting-started/setup/installation-guides/kubernetes/README.mdx:27]: we are recommending chat widget everwhere right?

File: website/docs/getting-started/setup/installation-guides/kubernetes/README.mdx [hunk 14]
```diff
@@ -11,72 +11,99 @@ This page provides steps to install Appsmith on a Kubernetes cluster using the [
...
 
-</TabItem>
-<TabItem label="Microk8s" value="Microk8s">
+    ```bash
+    kubectl cluster-info
+    ```
+    The above command provides a summary of the current cluster configuration, including the Kubernetes master and other cluster information. 
 
-  a. Set up Kubernetes by [working with kubectl](https://microk8s.io/docs/working-with-kubectl).
+    d. A default storage class running. If not created yet, create the storage class with:
```
Comment (danciaclara) [inline website/docs/getting-started/setup/installation-guides/kubernetes/README.mdx:52]:
```suggestion
    4. Define storage class. If you haven't already, define the storage class with:
```

File: website/docs/getting-started/setup/installation-guides/kubernetes/README.mdx [hunk 15]
```diff
@@ -11,72 +11,99 @@ This page provides steps to install Appsmith on a Kubernetes cluster using the [
...
 
-  b. Ensure that you have a default storage class running. If not, [enable storage class](https://microk8s.io/docs/command-reference#heading--microk8s-enable).
-    
-</TabItem>
-</Tabs>
+    ```bash
```
Comment (danciaclara) [inline website/docs/getting-started/setup/installation-guides/kubernetes/README.mdx:54]: if we are providing them commands, we should always specify where they have run these.

File: website/docs/getting-started/setup/installation-guides/kubernetes/publish-appsmith-online.mdx [hunk 16]
```diff
@@ -8,125 +8,89 @@ import TabItem from '@theme/TabItem';
 The page provides steps to publish Appsmith Kubernetes installation to the internet.
```
Comment (danciaclara) [inline website/docs/getting-started/setup/installation-guides/kubernetes/publish-appsmith-online.mdx:8]: i think we should change the title to "Publish K8s to Internet" or "Expose K8s to Internet" We don't need to specify Appsmith installation and all

**Review by danciaclara — CHANGES_REQUESTED** [review]

File: website/docs/getting-started/setup/installation-guides/kubernetes/README.mdx (hunk 13 above)
Comment (jnikhila) [reply website/docs/getting-started/setup/installation-guides/kubernetes/README.mdx:27]: Yes, but here it makes sense for them to contact support using email, as they are having issues in their prerequisite setup. What do you think?

File: website/docs/getting-started/setup/installation-guides/kubernetes/README.mdx (hunk 14 above)
Comment (jnikhila) [reply website/docs/getting-started/setup/installation-guides/kubernetes/README.mdx:52]: It's not defining storage class. They have to create it.

File: website/docs/getting-started/setup/installation-guides/kubernetes/README.mdx (hunk 3 above)
Comment (jnikhila) [reply website/docs/getting-started/setup/installation-guides/kubernetes/README.mdx:61]: @pratapaprasanna - this looks the same to me.

File: website/docs/getting-started/setup/installation-guides/kubernetes/publish-appsmith-online.mdx (hunk 16 above)
Comment (jnikhila) [reply website/docs/getting-started/setup/installation-guides/kubernetes/publish-appsmith-online.mdx:8]: Expose k8s to Internet

File: website/docs/getting-started/setup/installation-guides/kubernetes/README.mdx (hunk 15 above)
Comment (jnikhila) [reply website/docs/getting-started/setup/installation-guides/kubernetes/README.mdx:54]: Just looked at it again, all the above steps are commands only, and they don't need to do it in a specific folder. Just on the terminal. I think it's intuitive that the users will open a terminal and we don't have to specify that. @pratapaprasanna - Please confirm.

File: website/docs/getting-started/setup/installation-guides/kubernetes/README.mdx (hunk 13 above)
Comment (danciaclara) [reply website/docs/getting-started/setup/installation-guides/kubernetes/README.mdx:27]: either way its going to be answered by Pranav and team only. The reason I'm thinking we should use the chat widget is to be able to easily measure how many people have issues and reach out from docs. Secondly its one less step for them to send an email

**Review by danciaclara — APPROVED** [review]

File: website/docs/getting-started/setup/installation-guides/kubernetes/README.mdx (hunk 8 above)
Comment (pratapaprasanna) [reply website/docs/getting-started/setup/installation-guides/kubernetes/README.mdx:55]: **for Kubernetes Version <= 1.20
we can use this 
for Kubernetes Version >= 1.23**

```
helm repo add aws-ebs-csi-driver https://kubernetes-sigs.github.io/aws-ebs-csi-driver
helm repo update
helm upgrade --install aws-ebs-csi-driver \
    --namespace kube-system \
    aws-ebs-csi-driver/aws-ebs-csi-driver
```
**Verify the installation using** 
```
kubectl get pods -n kube-system -l app.kubernetes.io/name=aws-ebs-csi-driver
```

**Review by pratapaprasanna — APPROVED** [review]

**Review by danciaclara — APPROVED** [review]

**Pull request merged**